src/Combine.elm

```diff
@@ -248,11 +248,11 @@ manyWithEndLocationForLastElement defaultRange getRange (Parser p) =
             Core.oneOf
                 [ p oldState
                     |> Core.map (\( newState, item ) -> Core.Loop ( newState, item :: items ))
-                , Core.succeed ()
-                    |> Core.map
-                        (\() ->
-                            Core.Done ( oldState, ( endLocationForList defaultRange getRange items, List.reverse items ) )
-                        )
+                , Core.lazy
+                    (\() ->
+                        Core.succeed
```

Comment:
Hmm, I wonder if this is faster. As far as I know `Parser.lazy` is only there to avoid cyclic references between parsers, not to improve performance. Maybe it does as well but I'm not sure :thinking:

Reply:
map needs to check whether it's valid or not underneath, so this kind of makes sense to me

Reply:
Oh right, that's a good point. But to do that it also re-defines the parser everytime it gets evaluated, whereas with `map` the parser is already defined, and (from prior optimizations) that was quite bad for performance.

I really don't know what is faster though. I'm okay with taking this change, and then benchmarking this change to know what is faster, because this would be a cool learning.